CMakePresets.json

```diff
@@ -0,0 +1,59 @@
+{
+    "version"   : 3,
+    "cmakeMinimumRequired" : {
+        "major" : 3,
+        "minor" : 22,
+        "patch" : 0
+    },
+    "configurePresets" : [
+        {
+            "name"          : "default",
+            "binaryDir"     : "${sourceDir}/build-with-${presetName}",
+            "cacheVariables" : {
+                "CMAKE_EXPORT_COMPILE_COMMANDS"     : {"type" : "BOOL"  , "value" : "ON"},
+                "CMAKE_BUILD_TYPE"                  : {"type" : "STRING", "value" : "Release"},
+                "CMAKE_CXX_STANDARD"                : {"type" : "STRING", "value" : "17"},
+                "KokkosTools_REUSE_KOKKOS_COMPILER" : {"type" : "BOOL"  , "value" : "ON"},
+                "Kokkos_DIR"                        : {"type" : "STRING", "value" : "$penv{Kokkos_DIR}"}
```

Comment:
This is not the right option to help find Kokkos.  I'd expect one should use `Kokkos_ROOT`.
I suspect Kokkos is being found because CMake actually parses the environment for <PackageName>_DIR.
Either drop it or use Kokkos_ROOT.

Reply:
I think I got inspiration from https://github.com/kokkos/kokkos-tools/blob/3855d8f6186b600b97ae81cd1d9417ad8e34c776/example/CMakeLists.txt#L1-L10, but you're right!

Reply:
FWIW, it's totally fine to set `Kokkos_DIR` pointing to the directory holding `KokkosConfig.cmake`.

Reply:
Is the current solution with `Kokkos_ROOT` OK for you both @masterleinad and @dalg24 ?

Reply:
I guess it's OK for you guys :wink: